tests/test_gui.py

```diff
@@ -0,0 +1,194 @@
+
+
+@pytest.fixture(scope="session")
+def context():
+    # A task in Big Buck Bunny which we're going to use
+    # for the current context.
+    return {"type": "Project", "id": 70}
+
+
+# This fixture will launch tk-run-app on first usage
+# and will remain valid until the test run ends.
+@pytest.fixture(scope="session")
+def host_application(context):
+    """
+    Launch the host application for the Toolkit application.
+
+    TODO: This can probably be refactored, as it is not
+    likely to change between apps, except for the context.
+    One way to pass in a context would be to have the repo being
+    tested to define a fixture named context and this fixture
+    would consume it.
+    """
+    process = subprocess.Popen(
+        [
+            "python",
+            "-m",
+            "tk_toolchain.cmd_line_tools.tk_run_app",
+            # Allows the test for this application to be invoked from
+            # another repository, namely the tk-framework-widget repo,
+            # by specifying that the repo detection should start
+            # at the specified location.
+            "--location",
+            os.path.dirname(__file__),
+            "--context-entity-type",
+            context["type"],
+            "--context-entity-id",
+            str(context["id"]),
+        ]
+    )
+    try:
+        yield
+    finally:
+        # We're done. Grab all the output from the process
+        # and print it so that is there was an error
+        # we'll know about it.
+        stdout, stderr = process.communicate()
+        sys.stdout.write(stdout or "")
+        sys.stderr.write(stderr or "")
+        process.poll()
+        # If returncode is not set, then the process
+        # was hung and we need to kill it
+        if process.returncode is None:
+            process.kill()
+        else:
+            assert process.returncode == 0
+
+
+@pytest.fixture(scope="session")
+def app_dialog(host_application):
+    """
+    Retrieve the application dialog and return the AppDialogAppWrapper.
+    """
+    before = time.time()
+    while before + 30 > time.time():
+        if sgtk.util.is_windows():
+            app_dialog = AppDialogAppWrapper(topwindows)
+        else:
+            app_dialog = AppDialogAppWrapper(topwindows["python"])
+
+        if app_dialog.exists():
+            yield app_dialog
+            app_dialog.close()
+            return
+    else:
+        raise RuntimeError("Timeout waiting for the app dialog to launch.")
+
+
+class AppDialogAppWrapper(object):
+    """
+    Wrapper around the app dialog.
+    """
+
+    def __init__(self, parent):
+        """
+        :param root:
+        """
+        self.root = parent["Shotgun: Publish"].get()
+
+    def exists(self):
+        """
+        ``True`` if the widget was found, ``False`` otherwise.
+        """
+        return self.root.exists()
+
+    def close(self):
+        self.root.buttons["Close"].mouseClick()
+
+
+def test_app(app_dialog):
```

Comment:
Please add docstrings for each test methods so we know what is going on in each tests. Also, test_app is too simple a test name.